schedule/__init__.py

```diff
@@ -202,180 +202,197 @@ class Job(object):
     A job is usually created and returned by :meth:`Scheduler.every`
     method, which also defines its `interval`.
     """
-    def __init__(self, interval, scheduler=None):
-        self.interval = interval  # pause interval * unit between runs
-        self.latest = None  # upper limit to the interval
-        self.job_func = None  # the job job_func to run
-        self.unit = None  # time units, e.g. 'minutes', 'hours', ...
-        self.at_time = None  # optional time at which this job runs
-        self.last_run = None  # datetime of the last run
-        self.next_run = None  # datetime of the next run
-        self.period = None  # timedelta between runs, only valid for
-        self.start_day = None  # Specific day of the week to start on
-        self.tags = set()  # unique set of tags for the job
-        self.scheduler = scheduler  # scheduler to register with
-
-    def __lt__(self, other):
+
+    def __init__(self, interval: int, scheduler: Optional[Scheduler] = None):
+        self.interval: int = interval  # pause interval * unit between runs
+        self.latest: Optional[int] = None  # upper limit to the interval
+        self.job_func: Optional[Callable] = None  # the job job_func to run
+
+        # time units, e.g. 'minutes', 'hours', ...
+        self.unit: Optional[str] = None
+
+        # optional time at which this job runs
+        self.at_time: Optional[datetime.time] = None
+
+        # datetime of the last run
+        self.last_run: Optional[datetime.datetime] = None
+
+        # datetime of the next run
+        self.next_run: Optional[datetime.datetime] = None
+
+        # timedelta between runs, only valid for
+        self.period: Optional[datetime.timedelta] = None
+
+        # Specific day of the week to start on
+        self.start_day: Optional[str] = None
+        self.tags: Set[str] = set()  # unique set of tags for the job
+        self.scheduler: Optional[
+            Scheduler
+        ] = scheduler  # scheduler to register with
+
+    def __lt__(self, other) -> bool:
         """
         PeriodicJobs are sortable based on the scheduled time they
         run next.
         """
         return self.next_run < other.next_run
 
-    def __str__(self):
-        return (
-            "Job(interval={}, "
-            "unit={}, "
-            "do={}, "
-            "args={}, "
-            "kwargs={})"
-        ).format(self.interval,
-                 self.unit,
-                 self.job_func.__name__,
-                 self.job_func.args,
-                 self.job_func.keywords)
+    def __str__(self) -> str:
+        return ("Job(interval={}, unit={}, do={}, args={}, kwargs={})").format(
```

Comment:
These formatting changes are the result of using Black I guess? I'm not opposed to change the formatting in this pr, it would have been 'cleaner' to do the formatting change in a separate PR but since we are here now let's just continue it. To make the introduction of black complete, let's add it to `requirements-dev.txt` and update [the doc](https://github.com/dbader/schedule/blob/master/docs/development.rst) with a new paragraph on how to use this.

Reply:
I am completely with you! Black changes are rather nice to do. I could create a new PR for black. We could merge that one first, I'll re-base this PR and we merge it. What do you think about that?

Reply:
Sounds good!

Reply:
See #432 for the formatting pr